--- util-linux-ng-2.18/Makefile.am	2010-03-18 18:11:23.000000000 -0400
+++ util-linux-ng-2.18/Makefile.am	2011-02-01 23:56:33.000000000 -0500
@@ -2,18 +2,9 @@
 
 SUBDIRS = \
 	include \
-	disk-utils \
-	fdisk \
 	getopt \
-	lib \
 	shlibs \
-	login-utils \
-	misc-utils \
-	po \
-	schedutils \
-	sys-utils \
-	text-utils \
-	tests
+	po
 
 
 RCS_FIND_IGNORE := \( -name SCCS -o -name BitKeeper -o -name .svn -o \
@@ -21,7 +12,6 @@
 
 if LINUX
 SUBDIRS += \
-	hwclock \
 	partx
 endif
 
